src/server/game/Entities/Item/Item.cpp

```diff
@@ -962,6 +962,13 @@ bool Item::GemsFitSockets() const
     return true;
 }
 
+bool Item::HasSocket() const
+{
+    // There can only be one socket added, and it's always in slot `PRISMATIC_ENCHANTMENT_SLOT`.
+    //     Built-in sockets                        Socket from upgrade
+    return this->GetTemplate()->Socket[0].Color || this->GetEnchantmentId(EnchantmentSlot(PRISMATIC_ENCHANTMENT_SLOT));
```

Comment:
shouldn't we check that `Socket[0]` is not null before accessing `.Color` ?

Reply:
- `GetTemplate()` Can't be null, an item always have a template
- `->Socket` Can't be null, every template is created with 3 socket
- `Socket[0].Color` Can't dereference `nullptr` because `Socket[0]` is a value, not a pointer